updated cache doc

daminichopra opened the pull request:
## Proposed changes

Added description in cache backend section

## Types of changes

What types of changes does your code introduce to the project: _Put
an `x` in the boxes that apply_

- [ x ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
      functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after
creating the PR. If you're unsure about any of them, don't hesitate to
ask. We're here to help! This is simply a reminder of what we are going
to look for before merging your code._

- [ x ] I have read the [CONTRIBUTING][contrib] doc
- [ x ] I have signed the [CLA][cla]
- [ ] I have added tests that prove my fix is effective or that my
      feature works
- [ x ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in
      downstream modules

## Further comments

updated changes as per [Issue](https://github.com/orchestracities/ngsi-timeseries-api/issues/559)



[cla]: https://raw.githubusercontent.com/orchestracities/ngsi-timeseries-api/master/individual-cla.pdf
    "Martel Open Source Software Individual Contributor License Agreement"
[contrib]: https://github.com/smartsdk/ngsi-timeseries-api/blob/master/CONTRIBUTING.md
    "Contributing to QuantumLeap"

daminichopra commented:
I have read the CLA Document and I hereby sign the CLA

c0c0n3 commented on docs/manuals/index.md:
```diff
@@ -275,6 +275,40 @@ As of today, the query caching stores:
+- `WQ_SUCCESS_TTL` - `How long, in seconds, before removing successfully run
+tasks from the work queue.Default: 86400 (a day)`
+- `WQ_WORKERS` - `How many worker queue processors to spawn`
+
```
should we add a link to the config section where these vars are documented fully? e.g. further info about those vars in the [link to config].

daminichopra replied:
@c0c0n3 , Sure i will add link and update commit.

c0c0n3 commented on docs/manuals/index.md:
```diff
@@ -275,6 +275,40 @@ As of today, the query caching stores:
+**Geocoding cache**
+To enable this, need to pass:
+    `CACHE_GEOCODING`: `True or False enable or disable caching for geocoding`
+
```
I think @chicco785 wanted to flesh out this subsection, see #559
you can find some docs about geocoding in the geo-coding module:

- https://github.com/orchestracities/ngsi-timeseries-api/blob/master/src/geocoding/geocoding.py

Maybe we could sum up that? e.g. when an entity comes in if geo-coding is enabled and the entity has an address (detail a bit which fields we consider) then QL tries to look up the corresponding geo coords in open street maps. the OSM response gets cached in Redis to avoid further (expensive!) look ups in OSM...

daminichopra replied:
@c0c0n3 , I have update some info in file as per comment. Also, I have updated email for commit but CLA is getting failed again and again.

c0c0n3 replied:
fab! thank you sooo much!!!

c0c0n3 commented:
@daminichopra that's great, thanks alot!! there's just a couple of things we could improve on, see my comments above.

daminichopra commented:
recheckcla

daminichopra commented:
recheckcla

daminichopra commented:
> @daminichopra that's great, thanks alot!! there's just a couple of things we could improve on, see my comments above.

I have updated commit after some changes. Please let me know if any modification is required. Thank you

c0c0n3 commented on docs/manuals/index.md:
```diff
@@ -275,6 +275,61 @@ As of today, the query caching stores:
+name and postal number, it maps that to a point and hence the generated
+attribute will be of type geo:point. Without a postal number, the address
+will represent the street or the city boundaries or even the country boundaries.
+
```
@daminichopra can you add a sentence just to explain why this has anything to do w/ caching? e.g.

QuantumLeap uses Open Street Maps (OSM) to find the geo-location corresponding to the entity's address. This is usually a rather expensive call, thus a Redis cache is used to avoid looking up the same address over and over again over a short period of time---e.g. think a batch entity update containing several entities sharing the same address. To enable caching of geo-location data, you need to use the following environment variable:

c0c0n3 commented:
> I have updated commit after some changes. 

Good stuff!

> Please let me know if any modification is required

Yes, if you could please add a sentence to explain what caching has to do w/ geo-location. If you like you can just copy-paste the example sentence I added as a review comment.

Also, the makrdown linter is moaning about lines being too long:

```
docs/manuals/index.md:308:81 MD013/line-length Line length [Expected: 80; Actual: 181]
docs/manuals/index.md:331:81 MD013/line-length Line length [Expected: 80; Actual: 123]
```

If you could please fix those two lines above and make sure any other lines you're going to add get truncated at 79 chars. So we can keep the linter happy :-)

> I have updated email for commit but CLA is getting failed again and again.

I looked into that and it's really beyond me. The error message is so cryptic it really doesn't help. We're going to need some time to figure out why the CLA action is failing.

Thanks again for your contribs!!

daminichopra commented:
> > I have updated commit after some changes.
> 
> Good stuff!
> 
> > Please let me know if any modification is required
> 
> Yes, if you could please add a sentence to explain what caching has to do w/ geo-location. If you like you can just copy-paste the example sentence I added as a review comment.
> 
> Also, the makrdown linter is moaning about lines being too long:
> 
> ```
> docs/manuals/index.md:308:81 MD013/line-length Line length [Expected: 80; Actual: 181]
> docs/manuals/index.md:331:81 MD013/line-length Line length [Expected: 80; Actual: 123]
> ```
> 
> If you could please fix those two lines above and make sure any other lines you're going to add get truncated at 79 chars. So we can keep the linter happy :-)
> 
> > I have updated email for commit but CLA is getting failed again and again.
> 
> I looked into that and it's really beyond me. The error message is so cryptic it really doesn't help. We're going to need some time to figure out why the CLA action is failing.
> 
> Thanks again for your contribs!!

@c0c0n3 , I have Added a sentence in doc and fixed markdown linter issue.
Also, I am looking into cause of failing CLA issue. 
Thanks for suggestions for this PR.

chicco785 commented:
@daminichopra the cla issue was due to the fact that some of the commits where signed as "root", i fixed that

chicco785 commented:
@c0c0n3 test_reporter is still failing, which look extremely odd. 
- master builds without error
- test pr for verifying if there is some weird change in circle-ci, builds without error

i am clueless

daminichopra commented:
> @daminichopra the cla issue was due to the fact that some of the commits where signed as "root", i fixed th

> @c0c0n3 test_reporter is still failing, which look extremely odd.
> 
> * master builds without error
> * test pr for verifying if there is some weird change in circle-ci, builds without error
> 
> i am clueless

@chicco785 , I will check PR locally once

daminichopra commented:
> > > @daminichopra the cla issue was due to the fact that some of the commits where signed as "root", i fixed th
> > 
> > 
> > > @c0c0n3 test_reporter is still failing, which look extremely odd.
> > > 
> > > * master builds without error
> > > * test pr for verifying if there is some weird change in circle-ci, builds without error
> > > 
> > > i am clueless
> > 
> > 
> > @chicco785 , I will check PR locally once
> 
> locally all works for me...

Some tests are failing for master and PR IN my env
